backend/app/usecase/search/search.go

```diff
@@ -0,0 +1,36 @@
+package search
+
+import (
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+// Search fetches URLs and users from persistent
+// storage, such as database
+type Search struct {
+	userURLRepo repository.UserURLRelation
+	urlRepo     repository.URL
+}
+
+// SearchForURLs fetches all URLs for a given user
+func (s Search) SearchForURLs(user entity.User) ([]entity.URL, error) {
+	aliases, err := s.userURLRepo.FindAliasesByUser(user)
```

Comment:
I do concern about the possible high load problems. I don't know we have restrictions for a number of links created by a user, but how to be if a user has 10k links?

I suggest implementing `paginator` in order to make this potential problem less impact.
What are your thoughts about it?